nemo/collections/tts/data/text_to_speech_dataset.py

```diff
@@ -115,26 +115,31 @@ def __init__(
 
         if featurizers:
             logging.info(f"Found featurizers {featurizers.keys()}")
-            self.featurizers = featurizers.values()
+            self.featurizers = list(featurizers.values())
         else:
             self.featurizers = []
 
         if feature_processors:
             logging.info(f"Found featurize processors {feature_processors.keys()}")
-            self.feature_processors = feature_processors.values()
+            self.feature_processors = list(feature_processors.values())
         else:
             self.feature_processors = []
 
-        self.align_prior_config = align_prior_config
-        if self.align_prior_config.use_beta_binomial_interpolator:
-            self.beta_binomial_interpolator = BetaBinomialInterpolator()
+        if align_prior_config:
+            self.align_prior_config = AlignPriorConfig(**align_prior_config)
+            if self.align_prior_config.use_beta_binomial_interpolator:
```

Comment:
Since this a new dataset class, could we just drop the binomial interpolator?

I think we just forgot to deprecate it, it's been obsolete for some time now

Reply:
I think we can drop it. I did not want to remove it because I was having issues with alignment during testing. But I think I found the root cause being that me redefining an "epoch" as 1,000 steps was causing the aligner binarization loss to warm up too quickly on large datasets. Which can be fixed by reducing the scale on the binarization loss, or making 'bin_loss_warmup_epochs' larger.

I think it is similar to the alignment issues we were seeing with FastPitch recently where aligner would not learn properly during fine-tuning.